lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -282,97 +398,167 @@ def parse_genbank(self, file_path, params):
-            unmatched_ids = list()
-            unmatched_ids_md5s = list()
-            for current_contig in self.contig_seq.keys():
-                current_contig_md5 = hashlib.md5(
-                    str(self.contig_seq[current_contig]).encode('utf8')
-                ).hexdigest()
-                if current_contig in ret['data']['contigs']:
-                    if current_contig_md5 != ret['data']['contigs'][current_contig]['md5']:
-                        unmatched_ids_md5s.append(current_contig)
-                else:
-                    unmatched_ids.append(current_contig)
-            if len(unmatched_ids) > 0:
-                raise ValueError(warnings['assembly_ref_extra_contigs'].format(", ".join(unmatched_ids)))
-            if len(unmatched_ids_md5s) > 0:
-                raise ValueError(warnings["assembly_ref_diff_seq"].format(", ".join(unmatched_ids_md5s)))
-            logging.info(f"Using supplied assembly: {assembly_ref}")
-            return assembly_ref
-        logging.info("Saving sequence as Assembly object")
-        Bio.SeqIO.write(out_contigs, fasta_file, "fasta")
-        assembly_ref = self.aUtil.save_assembly_from_fasta(
-            {'file': {'path': fasta_file},
-             'workspace_name': params['workspace_name'],
-             'assembly_name': assembly_id,
-             'type': params.get('genome_type', 'isolate'),
-             'contig_info': extra_info})
-        logging.info(f"Assembly saved to {assembly_ref}")
-        return assembly_ref
+            genome_obj.contig_seq[in_contig.id] = in_contig.seq.upper()
+        return out_contigs, extra_info
+
+    def _get_contigs_and_validate_existing_assembly(self, genome_objs):
+        ref2genome = {}
+        for genome_obj in genome_objs:
+            assembly_ref = genome_obj.use_existing_assembly
+            if assembly_ref:
+                contigs = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
+                _, extra_info = self._get_contigs_and_extra_info(
+                    contigs, genome_obj
+                )
+                self._validate_existing_assembly(
+                    assembly_ref, genome_obj
+                )
+                ref2genome[assembly_ref] = genome_obj
+                genome_obj.extra_info = extra_info
+                genome_obj.assembly_ref = assembly_ref
+                genome_obj.assembly_path = None
+
+        if ref2genome:
+            assembly_refs = list(ref2genome.keys())
+            results = self._get_objects_data(assembly_refs)
+            for assembly_ref, result in zip(assembly_refs, results):
+                object_data = result["data"]
+                assembly_info = result["info"]
+                genome = ref2genome[assembly_ref]
+                genome.assembly_info = assembly_info
+                genome.gc_content = object_data["gc_content"]
+                genome.dna_size = object_data["dna_size"]
+                genome.md5 = object_data["md5"]
```

Comment:
This is pulling the entire object just to get those last 3 bits of info which are available via the metadata

Reply:
we need assembly info + last 3 bits of info from the metadata

Reply:
yes, but you don't need the full object to get that

Reply:
use ws.get_object_info3 now and factor common code in a private helper function.

Reply:
So `_get_objects_data` no longer gets the data so should probably be renamed

Reply:
👍